server/server.go

```diff
@@ -1022,92 +903,98 @@ func (s *server) LoadSnapshot(ctx context.Context, req *rpcpb.LoadSnapshotReques
 		snapshotsDir:        s.cfg.SnapshotsDir,
 	})
 	if err != nil {
-		s.network = nil
 		return nil, err
 	}
+	s.clusterInfo = &rpcpb.ClusterInfo{
+		Pid:         pid,
+		RootDataDir: rootDataDir,
+	}
 
 	// blocking load snapshot to soon get not found snapshot errors
-	if err := s.network.loadSnapshot(ctx, req.SnapshotName); err != nil {
+	if err := s.network.LoadSnapshot(req.SnapshotName); err != nil {
 		s.log.Warn("snapshot load failed to complete", zap.Error(err))
-		s.network = nil
+		s.stopAndRemoveNetwork()
 		return nil, err
 	}
 
-	// start non-blocking wait to load snapshot results
-	// the user is expected to poll cluster status
-	readyCh := make(chan struct{})
-	go s.network.loadSnapshotWait(ctx, readyCh)
-
 	// update cluster info non-blocking
 	// the user is expected to poll this latest information
 	// to decide cluster/subnet readiness
 	go func() {
-		s.waitChAndUpdateClusterInfo("local cluster", readyCh, true)
+		ctx, cancel := context.WithTimeout(context.Background(), waitForHealthyTimeout)
+		defer cancel()
+		err := s.network.AwaitHealthy(ctx)
```

Comment:
we want this error to be returned by WaitForHealthy

Reply:
Not sure what you mean

Reply:
https://github.com/ava-labs/avalanche-network-runner/pull/447#discussion_r1131592992